### PR TITLE
Bounds check mdb_bind_column()

### PR DESCRIPTION
**Pull request by mjwoodcock**
Return -1 to report out of bounds column number.

This prevents a segfault when passing an invalid column number to this function.

### CODE REVIEW

**Comment by evanmiller** [comment]
Should this go ahead and return the `MdbColumn` instead of an integer?

See note in #187 about checking the new return value in other places.

**Comment by mjwoodcock** [comment]
I think it should return the same type as mdb_bind_column_by_name().

I'm happy to change them both.

However, I think it would be better to have separate functions to get the MdbColumn, (mdb_get_column() and mdb_get_column_by_name()?).  This would help abstract out the implementation details MdbTableDef struct.  Currently callers have to call g_ptr_array_index() on the array of columns.

**Comment by evanmiller** [comment]
Well, bind_by_name is returning an integer index - so it's technically the same integer type, but the meaning is pretty different! Maybe return the same integer index from both functions?

I would be very happy moving away from the GLib API on the client side. Right now we export GLib symbols when compiling with `--disable-glib`, which could could some link-time conflicts if the client app is linked to GLib but libmdb is not.

**Comment by mjwoodcock** [comment]
I could return the integer index. Would it be better to keep mdb_sql_bind_column() return value to be either the column index or -1 (or a #defined INVALID_COLUMN value) on error to keep it the same as this function?

**Comment by evanmiller** [comment]
I think ideally `mdb_sql_bind_column()` would return the actual value from `mdb_bind_column_by_name`, since it's calling that function.

**Comment by evanmiller** [comment]
Is this ready?

**Comment by mjwoodcock** [comment]
Yes, I believe it is.

File: src/libmdb/data.c [hunk 1]
```diff
@@ -477,16 +488,24 @@ mdb_fetch_row(MdbTableDef *table)
...
 	unsigned int i;
+	int ret;
 	char *bound_values[MDB_MAX_COLS]; 
 
 	for (i=0;i<table->num_cols;i++) {
 		bound_values[i] = (char *) g_malloc(256);
-		mdb_bind_column(table, i+1, bound_values[i], NULL);
+		ret = mdb_bind_column(table, i+1, bound_values[i], NULL);
+		if (ret == -1) {
+			fprintf(stdout, "error binding column %d\n", i+1);
```
Comment (evanmiller) [inline src/libmdb/data.c:498]: Should this go to `stderr`?

Comment (mjwoodcock) [reply src/libmdb/data.c:498]: Yes, it should.  Hang on...

**Pull request merged**